src/app/components/dashboard-language-selectbox/dashboard-language-selectbox.coffee

```diff
@@ -31,7 +31,7 @@ angular.module 'mnoEnterpriseAngular'
             bodyText: 'mno_enterprise.templates.components.language_selectbox.confirm'
             headerTextExtraData: { name: locale.name }
             bodyTextExtraData: { name: locale.name }
-            actionCb: -> MnoeCurrentUser.update(settings: {locale: $scope.selectedLangKey})
+            actionCb: -> MnoeCurrentUser.update(settings: Object.assign(MnoeCurrentUser.user.settings, locale: $scope.selectedLangKey))  # WIll destroy the whole previous settings.
```

Comment:
Why do you need to do this?

Reply:
The previous call would reset the whole settings `settings` attribute. Same logic is applied on the currency settings save action.  I don't really understand how it could be improved, except by creating an API call for settings modification